src/functions/connect.ts

```diff
@@ -0,0 +1,134 @@
+import { deviceCodes, messageConstants } from '../constants';
+import { buildRequest, request } from '../shared/functions';
+import { doesFetchWalletsOnLoad } from '../shared/predicates';
+import {
+  getPubKeyBytes,
+  getSharedSecret,
+  parseWallets,
+} from '../shared/utilities';
+import {
+  validateBaseUrl,
+  validateDeviceId,
+  validateKey,
+} from '../shared/validators';
+import { aes256_decrypt, getP256KeyPairFromPub } from '../util';
+
+export async function connect ({
+  id,
+  client,
+}: ConnectRequestFunctionParams): Promise<boolean> {
+  const { deviceId, key, baseUrl } = validateConnectRequest({
+    deviceId: id,
+    key: client.key,
+    baseUrl: client.baseUrl,
+  });
+  const url = `${baseUrl}/${deviceId}`;
+
+  const payload = await encodeConnectRequest(key);
+
+  const response = await requestConnect(payload, url);
+
+  const { isPaired, fwVersion, activeWallets, ephemeralPub } =
+    await decodeConnectResponse(response, key);
+
+  client.deviceId = deviceId;
+  client.ephemeralPub = ephemeralPub;
+  client.url = `${client.baseUrl}/${deviceId}`;
```

Comment:
I was wondering where this got set... may be worth a comment in `client.ts`

Reply:
10eb9e0cd49a54c754f287b74e8e74d4b98cb628